pkg/secrethub/client.go

```diff
@@ -139,6 +157,19 @@ func NewClient(with ...ClientOption) (*Client, error) {
 		}
 	}
 
+	appName := os.Getenv("SECRETHUB_APP_INFO_NAME")
+	if appName != "" {
+		appVersion := os.Getenv("SECRETHUB_APP_INFO_VERSION")
+		topLevelAppInfo := &AppInfo{
+			Name:    appName,
+			Version: appVersion,
+		}
+		// Ignore app info from environment variable if name is invalid
+		if err = topLevelAppInfo.ValidateName(); err == nil {
```

Comment:
Do we want to ignore it or just filter out any unsupported characters? The latter could also help with version numbers: if there is something illegal in there, the version number still gets through.

Reply:
For the app name, I think I'd prefer the user agent not getting through over an automatically altered name that may not make any sense.